src/Model/TranslatableTrait.php

```diff
@@ -126,30 +124,36 @@ public function removeTranslationWithLocale(string $locale): void
         foreach ($translations as $translation) {
             if ($translation->getLocale() === $locale) {
                 $this->removeTranslation($translation);
+
+                $translation->setTranslatable(null);
```

Comment:
@SpartakusMd this line already exists in removeTranslation method, so there is no need to call it here.